src/Orckestra.Composer.Cart/Repositories/Order/OrderRepository.cs

```diff
@@ -295,5 +295,36 @@ public virtual Task<OrderFulfillmentState> AddShipmentFulfillmentMessagesAsync(A
 
             return OvertureClient.SendAsync(request);
         }
+
+        /// <summary>
+        /// Get Customer Ordered Products
+        /// </summary>
+        /// <param name="param"></param>
+        /// <returns></returns>
+        public async Task<GetCustomerOrderedProductsResponse> GetCustomerOrderedProductsAsync(GetCustomerOrderedProductsParam param)
+        {
+            var request = new GetCustomerOrderedProductsRequest
+            {
+                ScopeId = param.ScopeId,
+                CustomerId = param.CustomerId,
+                StartDate = param.StartDate,
+                EndDate = param.EndDate,
+                MinimumOrderedNumberOfTimes = param.MinimumOrderedNumberOfTimes
+            };
+
+            var cacheKey = CustomerOrderedProductsCacheKey(param.ScopeId, param.CustomerId);
+            return await CacheProvider.GetOrAddAsync(cacheKey, () => OvertureClient.SendAsync(request)).ConfigureAwait(false);
```

Comment:
when method has one await and return Task, you can remove this await and async in method definition